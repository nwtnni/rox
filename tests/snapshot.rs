use std::fs;

use rox::lex::Scanner;

#[test]
fn lex() {
    insta::glob!("lex/*.rox", |path| {
        let source = fs::read_to_string(path).unwrap();
        let mut scanner = Scanner::new(&source);
        let tokens = scanner.scan();
        insta::assert_debug_snapshot!(tokens);
    });
}
