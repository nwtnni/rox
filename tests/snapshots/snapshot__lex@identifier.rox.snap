---
source: tests/snapshot.rs
expression: tokens
input_file: tests/lex/identifier.rox
---
[
    Identifier(
        "abwer",
    ),
    For,
    Identifier(
        "te9",
    ),
]
